### PR TITLE
ARCv3: Workaround for llock/scond livelock with exponential backoff

### PR DESCRIPTION
**Pull request by pavelvkozlov**
Add option ARC_LLSC_BACKOFF which enables delayed retry for failed SCOND instruction, read core id and provide different delays before LLSC retry for each core. 
Also add extra delay in spinlock wait loop before next read to prevent the data cache line from getting dirty and to give the other core time to complete the conditional store instruction (scond).

### CODE REVIEW

File: arch/arc/include/asm/spinlock.h [hunk 1]
```diff
@@ -239,16 +239,27 @@ static inline void arch_write_unlock(arch_rwlock_t *rw)
 	"4: ; --- done ---			\n"				\
 
 #define SCOND_FAIL_RETRY_VARS							\
-	  ,[delay] "+&r" (delay), [tmp] "=&r"	(tmp)				\
+	  ,[delay] "+&r" (delay), [wait] "+&r" (wait), [tmp] "=&r" (tmp)	\
+
+#define SPINLOCK_WAIT_LOOP_WITH_DELAY						\
+	"1:	llock	%[val], [%[slock]]	\n"				\
+	"	brne	%[val], %[LOCKED], 11f	\n"	/* if LOCKED */		\
```
Comment (xxkent) [inline arch/arc/include/asm/spinlock.h]: /* If not LOCKED */

Comment (xxkent) [reply arch/arc/include/asm/spinlock.h:246]: I've reviewed it and think we can merge this.

**Comment by xxkent** [comment]
I've reviewed it and think we can merge this.

File: arch/arc/include/asm/spinlock.h (hunk 1 above)
Comment (pavelvkozlov) [reply arch/arc/include/asm/spinlock.h]: Resolved in new rebase, thank you.

**Comment by pavelvkozlov** [comment]
Thank you for your review. Merged.

**Pull request merged**